src/NuGetPackageVerifier/Rules/PackageVersionMatchesAssemblyVersionRule.cs

```diff
@@ -6,40 +6,86 @@
+                    typeof(AssemblyInformationalVersionAttribute).FullName,
                     StringComparison.Ordinal));
 
-            if (versionAttribute == null)
+            var assemblyInformationalNuGetVersion = new NuGetVersion(assemblyInformationalVersionAttribute.ConstructorArguments[0].Value.ToString());
+            if (!VersionEquals(_packageVersion, assemblyInformationalNuGetVersion))
             {
-                yield break;
+                yield return PackageIssueFactory.AssemblyInformationalVersionDoesNotMatchPackageVersion(
+                    currentFilePath,
+                    assemblyInformationalNuGetVersion,
+                    _packageVersion,
+                    _packageId);
             }
 
-            var assemblyVersion = versionAttribute.AttributeType.Module.Assembly.Name.Version;
-            if (!_packageVersion.Equals(assemblyVersion))
+            var assemblyFileVersionAttribute = assemblyAttributes.SingleOrDefault(a =>
+                a.AttributeType.FullName.Equals(
+                    typeof(AssemblyFileVersionAttribute).FullName,
+                    StringComparison.Ordinal));
+
+            var assemblyFileNuGetVersion = new NuGetVersion(assemblyFileVersionAttribute.ConstructorArguments[0].Value.ToString());
+            if (!VersionEquals(_packageVersion, assemblyFileNuGetVersion))
+            {
+                yield return PackageIssueFactory.AssemblyFileVersionDoesNotMatchPackageVersion(
+                    currentFilePath,
+                    assemblyFileNuGetVersion,
+                    _packageVersion,
+                    _packageId);
+            }
+
+            var assemblyVersion = assemblyInformationalVersionAttribute.AttributeType.Module.Assembly.Name.Version.ToString();
+            if (!_packageVersion.Version.ToString().Equals(assemblyVersion))
+            {
+                yield return PackageIssueFactory.AssemblyVersionDoesNotMatchPackageVersion(
+                    currentFilePath,
+                    assemblyVersion,
+                    _packageVersion.Version,
+                    _packageId);
+            }
+        }
+
+        private bool VersionEquals(NuGetVersion packageVersion, NuGetVersion assemblyNuGetVersion)
+        {
+            if (packageVersion == null)
+            {
+                throw new ArgumentNullException("packageVersion");
+            }
+
+            if (assemblyNuGetVersion == null)
             {
-                yield return PackageIssueFactory.AssemblyVersionDoesNotMatchPackageVersion(currentFilePath, assemblyVersion, _packageVersion, _packageId);
+                throw new ArgumentNullException("assemblyNuGetVersion");
             }
+
+            if (packageVersion.Major.Equals(assemblyNuGetVersion.Major) &&
+                packageVersion.Minor.Equals(assemblyNuGetVersion.Minor) &&
+                packageVersion.Patch.Equals(assemblyNuGetVersion.Patch))
+            {
```

Comment:
Might also want to add a comment somewhere here that we specifically ignore the "prerelease" part of the version because we're ok with that not matching.